src/providers/eth_provider/provider.rs

```diff
@@ -75,7 +79,12 @@ where
     }
 
     /// Returns a reference to the Starknet provider.
-    pub const fn starknet_provider(&self) -> &SP {
+    pub const fn starknet_provider(&self) -> &StarknetProvider<SP> {
+        &self.starknet_provider
+    }
+
+    /// Returns a reference to the underlying SP provider.
+    pub fn starknet_provider_inner(&self) -> &SP {
         &self.starknet_provider
     }
```

Comment:
this does the same

Reply:
Not exactly because there is `Deref` which is implemented for the `StarknetProvider` and which automatically derefs to the inner provider which is of type `SP` for the multiple places in the code where we need to access `SP` and not `StarknetProvider<SP>`.

If you prefer, to have something more explicit, in the `StarknetProvider` I could have something like:
```rust
pub fn provider(&self) -> &SP {
&self.provider
}
```

But this would force us to explicitly call this function in situations like:
```rust
eth_client
.eth_provider()
.starknet_provider()
.get_nonce(starknet_block_id, felt_address)
.await
.unwrap_or_default(),
```

where the deref would not be automatic anymore, which would force us to have
```rust
eth_client
.eth_provider()
.starknet_provider()
.provider()
.get_nonce(starknet_block_id, felt_address)
.await
.unwrap_or_default(),
```

I feel like the `Deref` is well adapted to the situation, wdyt?

Reply:
ah didn't notice the deref, you I think that works then